crates/pathfinder/src/cairo/ext_py/ser.rs

```diff
@@ -46,3 +50,273 @@ impl From<crate::core::Chain> for UsedChain {
         }
     }
 }
+
+/// Custom type for setting the serialization in stone, or at least same as python code.
+#[derive(Debug)]
+pub struct ContractUpdatesWrapper<'a>(&'a HashMap<ContractAddress, Vec<StorageDiff>>);
+
+impl<'a> From<Option<&'a crate::sequencer::reply::StateUpdate>> for ContractUpdatesWrapper<'a> {
+    fn from(u: Option<&'a crate::sequencer::reply::StateUpdate>) -> Self {
+        lazy_static::lazy_static! {
+            static ref EMPTY_MAP: HashMap<
+                ContractAddress,
+                Vec<StorageDiff>,
+            > = HashMap::new();
+        }
+
+        if let Some(u) = u {
+            ContractUpdatesWrapper(&u.state_diff.storage_diffs)
+        } else {
+            ContractUpdatesWrapper(&EMPTY_MAP)
+        }
```

Comment:
Interesting way of doing this, is there a reason `Option<&...>` didn't work?

Reply:
Fixed in next push.